processor/routingprocessor/config.go

```diff
@@ -107,12 +112,56 @@ const (
 
 // RoutingTableItem specifies how data should be routed to the different exporters
 type RoutingTableItem struct {
-	// Value represents a possible value for the field specified under FromAttribute. Required.
+	// Value represents a possible value for the field specified under FromAttribute.
 	Value string `mapstructure:"value"`
 
+	// Expression is a TQL condition / query on which signals routing is based.
+	Expression string `mapstructure:"expression"`
+
 	// Exporters contains the list of exporters to use when the value from the FromAttribute field matches this table item.
 	// When no exporters are specified, the ones specified under DefaultExporters are used, if any.
 	// The routing processor will fail upon the first failure from these exporters.
 	// Optional.
 	Exporters []string `mapstructure:"exporters"`
 }
+
+// rewriteRoutingItemsToTQL rewrites routing table entries based on resource
+// attributes routing to TQL.
+func rewriteRoutingEntriesToTQL(cfg *Config) Config {
+	if cfg.AttributeSource != resourceAttributeSource {
+		return *cfg
+	}
+	table := make([]RoutingTableItem, 0)
```

Comment:
Wouldn't the size of the new table always match the size of the previous one? If so, you can initialize the array with the size of the previous array and avoid resizing the array.